astrocut/footprint_cutouts.py

```diff
@@ -0,0 +1,325 @@
+        sr_list = s_reg.strip().split()
+        reg_type = sr_list[0].upper()
+
+        if reg_type == 'POLYGON':
+            ras = np.array(sr_list[1::2], dtype=float)
+            ras[ras < 0] = ras[ras < 0] + 360
+            decs = np.array(sr_list[2::2], dtype=float)
+            return SphericalPolygon.from_radec(ras, decs)
+        else:
+            raise ValueError('Unsupported S_Region type.')
+
+    return np.vectorize(ind_sregion_to_polygon)(s_region)
+
+
+@cached(cache=FFI_TTLCACHE, lock=Lock())
+def _get_caom_ffis(product: str):
+    """
+    Using vo-tap, fetch the footprint file from CAOM containing a dict of all FFIs and a 
+    polygon column that holds the s_regions as polygon points and vectors.
+
+    This is a temporary work-around until the footprint files are publicly accessible
+    on the cloud.
+    """
+    # Define the URL and parameters for the query
+    obs_collection = 'TESS' if product == 'SPOC' else 'HLSP'
+    target_name = 'TESS FFI' if product == 'SPOC' else 'TICA FFI'
+    url = 'https://mast.stsci.edu/vo-tap/api/v0.1/caom/sync'
+    params = {
+        'FORMAT': 'csv',
+        'LANG': 'ADQL',
+        'QUERY': 'SELECT obs_id, t_min, t_max, s_region, target_name, sequence_number '
+                'FROM dbo.ObsPointing '
+                f"WHERE obs_collection='{obs_collection}' AND dataproduct_type='image' AND target_name='{target_name}'"
+    }
+
+    # Send the GET request
+    response = requests.get(url, params=params)
+    response.raise_for_status()  # Raise HTTPError if response is not 200
+
+    # Load CSV data into a Table
+    all_ffis = Table.read(response.text, format='csv')
+    all_ffis.sort('obs_id')
+
+    # Convert regions to polygons
+    all_ffis['polygon'] = _s_region_to_polygon(all_ffis['s_region'])
+
+    return all_ffis
+
+
+@cached(cache=FFI_TTLCACHE, lock=Lock())
+def _get_s3_ffis(s3_uri, as_table: bool = False, load_polys: bool = False):
+    """
+    Fetch the S3 footprint file containing a dict of all FFIs and a polygon column
+    that holds the s_regions as polygon points and vectors.
+
+    Optional Parameters:
+        as_table: Return the footprint file as an Astropy Table
+        load_polys: Convert the s_region column to an array of SphericalPolygon objects
+    """
+    # Open footprint file with fsspec
+    with fsspec.open(s3_uri, s3={'anon': True}) as f:
+        ffis = json.load(f)
+
+    if load_polys:
+        ffis['polygon'] = _s_region_to_polygon(ffis['s_region'])
+
+    if as_table:
+        ffis = Table(ffis)
+
+    return ffis
+
+
+def _ffi_intersect(ffi_list: Table, polygon: SphericalPolygon):
+    """
+    Vectorizing the spherical_coordinate intersects_polygon function
+    """
+    def single_intersect(ffi, polygon):
+        return ffi.intersects_poly(polygon)
+
+    return np.vectorize(single_intersect)(ffi_list['polygon'], polygon)
+
+
+def _ra_dec_crossmatch(all_ffis: Table, coord: SkyCoord, cutout_size, arcsec_per_px: int):
```

Comment:
I think it might be useful to provide this function (or one that gets the footprint for you as well) as an external interface. I could see this being useful for people who want more control over the cube_cut.

Could be done in a separate PR or issue if you don't want to do it here.